frontend/app/mocks/subscription-api.server.ts

```diff
@@ -74,5 +84,35 @@ export function getSubscriptionApiMockHandlers() {
 
       return HttpResponse.text(null, { status: 204 });
     }),
+    //MOCK TO VALIDATE THE CONFIRMATION CODE VS THE ONE ENTERED BY THE USER
+    http.post('https://api.example.com/v1/users/codes/verify', async ({ params, request }) => {
+      log.debug('Handling request for [%s]', request.url);
+      const timeEntered = new Date();
+      const requestBody = await request.json();
+      const validateSubscriptionSchemaData = validateSubscriptionSchema.safeParse(requestBody);
+      if (!validateSubscriptionSchemaData.success) {
+        throw new HttpResponse(null, { status: 400 });
+      }
+      const subscriptionConfirmationCodesEntities = db.subscriptionConfirmationCode.findMany({
+        where: { email: { equals: validateSubscriptionSchemaData.data.email } },
+      });
+
+      if (subscriptionConfirmationCodesEntities.length == 0) {
+        return HttpResponse.text(httpResponseMessages.NoCodeFound, { status: 200 });
+      }
+
+      const latestConfirmCode = subscriptionConfirmationCodesEntities.reduce((prev, current) => (prev.createdDate > current.createdDate ? prev : current));
+
+      if (latestConfirmCode.confirmationCode == validateSubscriptionSchemaData.data.confirmationCode && timeEntered < latestConfirmCode.expiryDate) {
+        return HttpResponse.text(httpResponseMessages.ValidCodeSuccess, { status: 200 });
+      }
+      if (latestConfirmCode.confirmationCode == validateSubscriptionSchemaData.data.confirmationCode && timeEntered > latestConfirmCode.expiryDate) {
+        //Code expired
+        return HttpResponse.text(httpResponseMessages.CodeExpired, { status: 200 });
+      }
+
+      //There is at least 1 confirmation code and the code entered by the user does not match it..
+      return HttpResponse.text(httpResponseMessages.CodeMismatch, { status: 200 });
```

Comment:
**Nitpick:** As a developer, I would have preferred reading the response messages inline and removing the need for the `httpResponseMessages` enum because I don't want to be scrolling up to the top of this file to see what the message actually is.

Reply:
The response should actually be in JSON format rather than a single message, maybe something like

```json
{ 
 "status": "mismatch"
}
```

Reply:
enum has been removed